portability/file.cc

```diff
@@ -277,25 +277,29 @@ toku_os_pwrite (int fd, const void *buf, size_t len, toku_off_t off) {
     return result;
 }
 
-FILE * 
+TOKU_FILE *
 toku_os_fdopen(int fildes, const char *mode) {
     FILE * rval;
     if (t_fdopen)
 	rval = t_fdopen(fildes, mode);
     else 
 	rval = fdopen(fildes, mode);
-    return rval;
+    if (FT_UNLIKELY(rval == nullptr))
+        return nullptr;
+    return new TOKU_FILE(rval);
 }
     
 
-FILE *
+TOKU_FILE *
 toku_os_fopen(const char *filename, const char *mode){
     FILE * rval;
     if (t_fopen)
 	rval = t_fopen(filename, mode);
     else
 	rval = fopen(filename, mode);
-    return rval;
+    if (FT_UNLIKELY(rval == nullptr))
+        return nullptr;
```

Comment:
Note that this function will return nullptr on failure, not a TOKU_FILE with the FILE* component NULL...